packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/TDSCodeController.java

```diff
@@ -148,6 +183,18 @@ private ResponseEntity<Integer> uploadCodeHelper(String codeId, String fileName,
 			put.setEntity(codeHttpEntity);
 			final HttpResponse response = httpclient.execute(put);
 
+			final Code code = codeProxy.getAsset(codeId).getBody();
+			final CodeFile codeFile = new CodeFile();
+			codeFile.setProgrammingLanguageFromFileName(fileName);
+
+			Map<String, CodeFile> fileMap = code.getFiles();
+
+			if(fileMap == null){
+				fileMap = new HashMap<>();
+			}
+			fileMap.put(fileName, codeFile);
+			code.setFiles(fileMap);
+			codeProxy.updateAsset(codeId, convertObjectToSnakeCaseJsonNode(code)).getBody();
```

Comment:
need to update the files object when we upload a file, we append to the map and then update the asset